internal/cmd/commands/scopescmd/rotate_keys.go

```diff
@@ -0,0 +1,118 @@
+package scopescmd
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/hashicorp/boundary/api"
+	"github.com/hashicorp/boundary/api/scopes"
+	"github.com/hashicorp/boundary/internal/cmd/base"
+	"github.com/mitchellh/cli"
+	"github.com/mitchellh/go-wordwrap"
+	"github.com/posener/complete"
+)
+
+var (
+	_ cli.Command             = (*RotateKeysCommand)(nil)
+	_ cli.CommandAutocomplete = (*RotateKeysCommand)(nil)
+)
+
+type RotateKeysCommand struct {
+	*base.Command
+}
+
+func (c *RotateKeysCommand) Synopsis() string {
+	return wordwrap.WrapString("Rotate the keys within a scope", base.TermWidth)
+}
+
+func (c *RotateKeysCommand) Help() string {
+	return base.WrapForHelpText([]string{
+		"Usage: boundary scopes rotate-keys [args]",
+		"",
+		"  Generate new keys for a scope and all nested scopes. Example:",
+		"",
+		`    $ boundary scopes rotate-keys -scope-id global`,
+		"",
+		"",
+	}) + c.Flags().Help()
+}
+
+func (c *RotateKeysCommand) Flags() *base.FlagSets {
+	// what are these and what do they do??
+	set := c.FlagSet(base.FlagSetHTTP | base.FlagSetClient | base.FlagSetOutputFormat)
+	f := set.NewFlagSet("Command Options")
+
+	f.StringVar(&base.StringVar{
+		Name:   "scope-id",
+		Target: &c.FlagScopeId,
+		Usage:  "The id of the scope in which to rotate keys",
+	})
+
+	f.BoolVar((&base.BoolVar{
+		Name:   "rewrap",
+		Target: &c.FlagRewrap,
+		Usage:  "Whether or not child keys should be re-encrypted by the newly generated keys",
```

Comment:
I feel like this could be worded a bit differently. How about
```suggestion
		Usage:  "Whether or not to re-encrypt DEKs with the new KEK",
```
Looks like this is the terminology we use externally: https://www.boundaryproject.io/docs/concepts/security/data-encryption#the-root-kms-key-and-per-scope-kek-deks-oss-only